collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -594,46 +597,22 @@ private void loadFromIntent(Intent intent) {
             }
 
             instancePath = formInfo.getInstancePath();
+            List<Form> candidateForms = formsRepository.getAll(formInfo.getFormId(), formInfo.getFormVersion());
```

Comment:
Oh wow it feels like this is a big improvement. I'm excited to get this stuff out of the Activity...